Shared/LayerCacheManager.cpp

```diff
@@ -442,7 +445,43 @@ void LayerCacheManager::connectSignals()
 
       for (int i = 0; i < layerCount; i++)
       {
-        m_scene->operationalLayers()->append(m_initialLayerCache.value(i));
+        auto* lyr = m_initialLayerCache.value(i);
+        if (!lyr)
+          continue;
+
+        if (auto* rasterLayer = dynamic_cast<RasterLayer*>(lyr); rasterLayer)
+        {
+          connect(rasterLayer, &RasterLayer::doneLoading, this, [this, rasterLayer](const Error& loadError)
+          {
+            // not much we can do here if the operational layers pointer is null...
+            auto* operationalLayers = ToolResourceProvider::instance()->operationalLayers();
+            if (!operationalLayers)
+              return;
+
+            auto rasterPath = rasterLayer->raster()->path();
+            if (!loadError.isEmpty())
+            {
+              operationalLayers->removeOne(rasterLayer);
+              rasterLayer->raster()->deleteLater();
+              rasterLayer->deleteLater();
+              emit toolErrorOccurred(QString("The raster (%1), failed to load. [%2][%3]").arg(rasterPath, loadError.message(), loadError.additionalMessage()), rasterFailedAdditionalMsg);
+              return;
+            }
+
+            // check the spatial reference for the new layer for:
+            // - non-georeferenced PDFs may have been added
+            if (const auto spatialReference = rasterLayer->spatialReference(); spatialReference.isEmpty())
```

Comment:
You don't need to hold onto the object at all here. 

```suggestion
            if (rasterLayer->spatialReference().isEmpty())
```

Reply:
Good catch. Yep, not using this anywhere else below. Can't remember why I was storing a copy of this to begin with so I'll switch it out.